core/src/test/java/org/infinispan/tx/locking/LocalPessimisticTxTest.java

```diff
@@ -31,6 +42,76 @@ public void testLockingWithRollback() throws Exception {
       assertNull(cache().get("k"));
    }
 
+   /*
+    * Reproducer for ISPN-13024
+    *
+    * The "real world" conditions are:
+    *  - Narayan timeout <= lock timeout
+    *
+    * Cause/Steps:
+    *  - Narayan aborts a transaction which is waiting for a lock "L"
+    *  - The RollbackCommand is sent.
+    *    - Lock "L" is not locked by the transaction, so unlockAll(context.getLockedKeys(), context.getLockOwner());
+    *      never changes the state from WAITING => RELEASED
+    *  - When the lock "L" is released, the state changes from WAITING => ACQUIRED
+    *  - No RollbackCommand/CommitCommand is followed leaving the lock "L" in ACQUIRED state forever
+    */
+   public void testReaperAbortsTransaction() throws Exception {
+      final String key = "_reaper_";
+      final String anotherKey = "_reaper_2_";
+      final CompletableFuture<TxData> txData = new CompletableFuture<>();
+
+      // simulates lock acquired by other transaction
+      lockManager().lock(key, "_tx_", 1, TimeUnit.SECONDS).lock();
+      assertLocked(cache, key);
+
+      // start a transaction and try to acquire the lock. The thread will block
```

Comment:
The thread only blocks because of `res.join()`, you could do everything on the main thread IMO.

Reply:
:+1: